terraform/variables/waypoint.hcl

```diff
@@ -0,0 +1,74 @@
+project = "waypoint_tfc_vars_example"
+
+app "waypoint_tfc_vars_example" {
+  build {
+    use "docker" {
+    }
+  }
+
+  deploy {
+    use "docker" {
+
+      labels = {
+        // `simple_string_output` matches the terraform output `simple_string`
+        "simple_string" = var.single_string_output
+
+        // `single_list_output` matches the terraform output `list_of_strings`.
+        // Note that the output is a list type in terraform, and can also be
+        // treated as a list here in the waypoint hcl.
+        "list_item"     = var.single_list_output[1]
+      }
+
+      // This shows referencing one variable after reading in all terraform outputs.
+      // the waypoint variable is `all_outputs`, and the tfc output name is `map_of_strings`
+      // (see ./terraform/main/outputs.tf).
+      // Also note that the expected type for `static_environment` is also a map of strings,
+      // so we can use the whole variable here. With this setting, the container will have
+      // environment variables for each k/v pair in `map_of_strings`.
+      static_environment = var.all_outputs.map_of_strings
```

Comment:
Sick!!! ✨

Reply:
I was so amazed when this _just worked_ after making the variable available to the HCL context. HCL is magic.